sdk/include/opentelemetry/sdk/trace/span_data.h

```diff
@@ -84,6 +84,35 @@ struct AttributeConverter
   }
 };
 
+/**
+ * Class for storing events in SpanData.
+ */
+class SpanDataEvent
+{
+public:
+  SpanDataEvent(nostd::string_view name, core::SystemTimestamp timestamp)
+  {
+    name_      = name;
+    timestamp_ = timestamp;
+  }
+
+  /**
+   * Get the name for this event
+   * @return the name for this event
+   */
+  nostd::string_view GetName() const noexcept { return name_; }
+
+  /**
+   * Get the timestamp for this event
+   * @return the timestamp for this event
+   */
+  core::SystemTimestamp GetTimestamp() const noexcept { return timestamp_; }
+
+private:
+  nostd::string_view name_;
```

Comment:
We have to store this as a `std::string`. Otherwise we might run into lifetime issues, as `string_view` is a non-owning reference.